components/script/dom/gpubuffer.rs

```diff
@@ -47,11 +48,12 @@ pub enum GPUBufferState {
 #[derive(JSTraceable, MallocSizeOf)]
 pub struct GPUBufferMapInfo {
     #[ignore_malloc_size_of = "Rc"]
-    pub mapping: Rc<RefCell<Vec<u8>>>,
+    #[no_trace]
```

Comment:
`#[ignore_malloc_size_of = "Arc"]`

Reply:
Actually it's different, sorry. Can you pls look in dom/bindings/trace, we should have implemented the right trait, but maybe not for this type..

Reply:
Even after adding trace implementation:

```
unsafe impl<T: JSTraceable> CustomTraceable for Arc<Mutex<Vec<T>>> {
    unsafe fn trace(&self, trc: *mut JSTracer) {
        for e in (**self).lock().unwrap().iter() {
            e.trace(trc);
        }
    }
}
``
I still get:

```
the method `trace` exists for reference `&Arc<Mutex<Vec<u8>>>`, but its trait bounds were not satisfied
the following trait bounds were not satisfied:
`std::sync::Mutex<std::vec::Vec<u8>>: js::rust::Trace`
which is required by `std::sync::Arc<std::sync::Mutex<std::vec::Vec<u8>>>: js::rust::Trace`
items from traits can only be used if the trait is in scop
```

Reply:
This worked:

```
unsafe impl<T: crate::dom::bindings::trace::JSTraceable> CustomTraceable
    for std::sync::Arc<std::sync::Mutex<Vec<T>>>
{
    unsafe fn trace(&self, trc: *mut js::jsapi::JSTracer) {
        for e in (**self).lock().unwrap().iter() {
            e.trace(trc);
        }
    }
}
```

Reply:
Ok had a look at other uses of mutex in script, and I agree with @sagudev we should just use both 
```
#[ignore_malloc_size_of = "Arc"]
#[no_trace]
```

instead of implementing the trait.